src/licensedcode/plugin_license.py

```diff
@@ -116,3 +117,57 @@ def get_scanner(
             license_text_diagnostics=license_text_diagnostics,
             license_url_template=license_url_template
         )
+
+    def process_codebase(self, codebase, **kwargs):
+        
+        if codebase.has_single_resource:
+            return
+
+        for resource in codebase.walk(topdown=False):
+            match_reference_license(resource,codebase)
+
+
+def match_reference_license(resource, codebase):
+    """
+    Return the updated license matches having reference to another files
```

Comment:
We need to improve this doc... this could be made clearer?
```suggestion
    Return the ``resource`` Resource updating and saving it in place, after adding new
    license matches (licenses and license_expressions) following their Rule
    ``referenced_filenames`` if any. Return None if this is not a file Resource.
```